.github/workflows/compile-queries.yml

```diff
@@ -0,0 +1,57 @@
+name: "Compile all queries using the latest stable CodeQL CLI"
+
+on:
+  push:
+    branches: [main] # makes sure the cache gets populated
+  pull_request:
+    branches:
+      - main
+      - "rc/*"
+
+jobs:
+  compile-queries:
+    runs-on: ubuntu-latest-xl
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      # calculate the merge-base with main, in a way that works both on PRs and pushes to main. 
+      - name: Calculate merge-base
```

Comment:
Getting the merge-base is needed to get a good cache-key, so yes, it's needed. 

I wanted something that I was sure worked on both PRs and on push to `main`, and I didn't find a magic property (like `github.sha`) that fit my needs, so that's why I ended up this approach.